src/rule.jl

```diff
@@ -316,6 +316,38 @@ macro rule(expr)
     end
 end
 
+"""
+    @capture ex LHS
+
+Uses a `Rule` object to capture an expression if it matches the LHS pattern. Returns `true` and injects
+slot variable match results into the calling scope when the LHS matches, otherwise returns false. The
+rule language for specifying the LHS pattern is the same in @capture as it is in `@rule`. Contextual matching
+is not yet supported
```

Comment:
```suggestion
Uses a `Rule` object to capture an expression if it matches the `pattern`. Returns `true` and injects
slot variable match results into the calling scope when the `pattern` matches, otherwise returns false. The
rule language for specifying the `pattern` is the same in @capture as it is in `@rule`. Contextual matching
is not yet supported

e.g.

    julia> @syms a; ex = a ^ a;

    julia> if @capture ex (~x) + (~x)
               @show x
           elseif @capture ex (~y) ^ (~y)
               @show y
           end;
    y = a
```
I think adding an example like this might help motivate the idea behind the macro, and show people why it's useful for it to return a `Bool`. Not sure if this is an ideal example, but it's the first that came to mind.